### PR TITLE
Add Enterprise IDP to Org management feature

### PR DESCRIPTION
**Pull request by kavinda1995**
- New template for the IDP
- Create modal for IDP creation
- Connect with the backend API
- Label to identify Organization IDPs in IDP list view
- Hide settings tab for Organization IDPs

### Related Issues
- https://github.com/wso2/product-is/issues/14304

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

**Review by thivi — APPROVED** [review]

**Comment by thivi** [comment]
Have we made sure that this won't show up on the cloud product?

File: apps/console/src/features/identity-providers/components/identity-provider-list.tsx [hunk 1]
```diff
@@ -292,47 +292,61 @@ export const IdentityProviderList: FunctionComponent<IdentityProviderListPropsIn
...
-                        <Header.Content>
-                            { idp.name }
-                            <Header.Subheader data-testid={ `${ testId }-item-sub-heading` }>
-                                { idp.description }
-                            </Header.Subheader>
-                        </Header.Content>
-                    </Header>
-                ),
+                render: (idp: IdentityProviderInterface): ReactNode => {
+                    const isOrgIdp = (idp.federatedAuthenticators.defaultAuthenticatorId ===
```
Comment (savindi7) [inline apps/console/src/features/identity-providers/components/identity-provider-list.tsx:296]: Shall we type this const

File: apps/console/src/features/identity-providers/components/wizards/organization-enterprise/organization-enterprise-authentication-provider-create-wizard-content.tsx [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+     * Callback to be triggered for form submit.
+     * @param values
+     */
+    onSubmit: (values: OrganizationEnterpriseAuthenticationProviderCreateWizardFormErrorValidationsInterface) => void;
+}
+
+/**
+ * Authentication Provider Create Wizard content component.
+ *
+ * @param {OrganizationEnterpriseAuthenticationProviderCreateWizardContentPropsInterface} props - Props injected to the component.
```
Comment (savindi7) [inline apps/console/src/features/identity-providers/components/wizards/organization-enterprise/organization-enterprise-authentication-provider-create-wizard-content.tsx:71]: Let's fix max-len issue here

File: apps/console/src/features/identity-providers/components/wizards/organization-enterprise/organization-enterprise-authentication-provider-create-wizard-content.tsx [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+            ["data-componentid"]: componentId
+        } = props;
+
+        const { t } = useTranslation();
+
+        const [ idpList, setIdPList ] = useState<IdentityProviderListResponseInterface>({});
+        const [ isIdPListRequestLoading, setIdPListRequestLoading ] = useState<boolean>(undefined);
+
+        /**
+     * Loads the identity provider authenticators on initial component load.
```
Comment (savindi7) [inline apps/console/src/features/identity-providers/components/wizards/organization-enterprise/organization-enterprise-authentication-provider-create-wizard-content.tsx:96]: Do we need a doc comment for this? Suggestion to use a single line comment instead.

File: apps/console/src/features/identity-providers/components/wizards/organization-enterprise/organization-enterprise-authentication-provider-create-wizard-content.tsx [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+        /**
+     * Check whether IDP name is already exist or not.
+     *
+     * @param value IDP name - IDP Name.
+     *
+     * @returns error msg if name is already taken.
+     */
+        const idpNameValidation = (value): string => {
+
+            let nameExist = false;
```
Comment (savindi7) [inline apps/console/src/features/identity-providers/components/wizards/organization-enterprise/organization-enterprise-authentication-provider-create-wizard-content.tsx:131]: Shall we type this?

File: apps/console/src/features/identity-providers/components/wizards/organization-enterprise/organization-enterprise-authentication-provider-create-wizard-content.tsx [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+            if (!values.description) {
+                errors.description = t("console:develop.features.authenticationProvider.forms.common" +
+                ".requiredErrorMessage");
+            }
+
+            return errors;
+        };
+
+        return (
+            (isIdPListRequestLoading !== undefined && isIdPListRequestLoading === false)
```
Comment (savindi7) [inline apps/console/src/features/identity-providers/components/wizards/organization-enterprise/organization-enterprise-authentication-provider-create-wizard-content.tsx:176]: `isIdPListRequestLoading !== undefined` is this part necessary?

File: apps/console/src/features/identity-providers/components/wizards/organization-enterprise/organization-enterprise-authentication-provider-create-wizard.tsx [hunk 6]
```diff
@@ -0,0 +1,484 @@
...
+
+        // Trigger the close method from props.
+        onWizardClose();
+    };
+
+    /**
+     * Close the limit reached modal.
+     */
+    const handleLimitReachedModalClose = (): void => {
+        setOpenLimitReachedModal(false);
```
Comment (kayathiri4) [inline apps/console/src/features/identity-providers/components/wizards/organization-enterprise/organization-enterprise-authentication-provider-create-wizard.tsx:255]: Add a new line.

File: apps/console/src/features/identity-providers/components/wizards/organization-enterprise/organization-enterprise-authentication-provider-create-wizard-content.tsx (hunk 2 above)
Comment (kavinda1995) [reply apps/console/src/features/identity-providers/components/wizards/organization-enterprise/organization-enterprise-authentication-provider-create-wizard-content.tsx:71]: Fixed in later commits

File: apps/console/src/features/identity-providers/components/wizards/organization-enterprise/organization-enterprise-authentication-provider-create-wizard-content.tsx (hunk 3 above)
Comment (kavinda1995) [reply apps/console/src/features/identity-providers/components/wizards/organization-enterprise/organization-enterprise-authentication-provider-create-wizard-content.tsx:96]: Used the same conventions used for other functions and useEffect hooks

File: apps/console/src/features/identity-providers/data/identity-provider-templates/templates/organization-enterprise-identity-provider/create-wizard-help.tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+/**
+ * Prop types of the component.
+ */
+type OrganizationEnterpriseIdentityProviderCreateWizardHelpPropsInterface = TestableComponentInterface
+
+/**
+ * Help content for the custom IDP template creation wizard.
+ *
+ * @param {OrganizationEnterpriseIdentityProviderCreateWizardHelpPropsInterface} props - Props injected into the component.
```
[the remark this answers is not in the document]
Comment (kavinda1995) [reply apps/console/src/features/identity-providers/data/identity-provider-templates/templates/organization-enterprise-identity-provider/create-wizard-help.tsx:32]: Fixed in later commit

File: apps/console/src/features/identity-providers/components/identity-provider-list.tsx (hunk 1 above)
Comment (kavinda1995) [reply apps/console/src/features/identity-providers/components/identity-provider-list.tsx:296]: Fixed in https://github.com/wso2/identity-apps/pull/3176

File: apps/console/src/features/identity-providers/components/wizards/organization-enterprise/organization-enterprise-authentication-provider-create-wizard-content.tsx (hunk 4 above)
Comment (kavinda1995) [reply apps/console/src/features/identity-providers/components/wizards/organization-enterprise/organization-enterprise-authentication-provider-create-wizard-content.tsx:131]: Fixed in https://github.com/wso2/identity-apps/pull/3176

File: apps/console/src/features/identity-providers/components/wizards/organization-enterprise/organization-enterprise-authentication-provider-create-wizard-content.tsx (hunk 5 above)
Comment (kavinda1995) [reply apps/console/src/features/identity-providers/components/wizards/organization-enterprise/organization-enterprise-authentication-provider-create-wizard-content.tsx:176]: Fixed in https://github.com/wso2/identity-apps/pull/3176

File: apps/console/src/features/identity-providers/data/identity-provider-templates/templates/organization-enterprise-identity-provider/create-wizard-help.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (kavinda1995) [reply apps/console/src/features/identity-providers/data/identity-provider-templates/templates/organization-enterprise-identity-provider/create-wizard-help.tsx:32]: Fixed in https://github.com/wso2/identity-apps/pull/3176

File: apps/console/src/features/identity-providers/data/identity-provider-templates/templates/organization-enterprise-identity-provider/create-wizard-help.tsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+        const {
+            ["data-testid"]: testId
+        } = props;
+
+        return (
+            <>
+                <div data-testid={ testId }>
+                    <Heading as="h5">Name</Heading>
+                    <p>Provide a unique name for the enterprise authentication provider so that it can be
```
[the remark this answers is not in the document]
Comment (kavinda1995) [reply apps/console/src/features/identity-providers/data/identity-provider-templates/templates/organization-enterprise-identity-provider/create-wizard-help.tsx:48]: Fixed in https://github.com/wso2/identity-apps/pull/3176

File: apps/console/src/features/identity-providers/components/wizards/organization-enterprise/organization-enterprise-authentication-provider-create-wizard.tsx (hunk 6 above)
Comment (kavinda1995) [reply apps/console/src/features/identity-providers/components/wizards/organization-enterprise/organization-enterprise-authentication-provider-create-wizard.tsx:255]: Fixed in https://github.com/wso2/identity-apps/pull/3176

**Pull request merged**